courses-rest/src/main/java/no/sikt/nva/fs/CoursesByInstitutionOfLoggedInUserHandler.java

```diff
@@ -1,43 +1,89 @@
+import nva.commons.core.paths.UriWrapper;
 
-public class CoursesByInstitutionOfLoggedInUserHandler extends ApiGatewayHandler<Void, CoursesByInstitutionResponse> {
+public class CoursesByInstitutionOfLoggedInUserHandler extends ApiGatewayHandler<Void, Course[]> {
 
-    private static final Logger logger = LoggerFactory.getLogger(CoursesByInstitutionOfLoggedInUserHandler.class);
+    /* default */ static final String FS_CONFIG_ENV_NAME = "FS_CONFIG";
+
+    private final TimeProvider timeProvider;
 
     @JacocoGenerated
     public CoursesByInstitutionOfLoggedInUserHandler() {
-        this(new Environment());
+        this(new Environment(), new SystemTimeProvider());
     }
 
-    public CoursesByInstitutionOfLoggedInUserHandler(final Environment environment) {
+    public CoursesByInstitutionOfLoggedInUserHandler(final Environment environment, final TimeProvider timeProvider) {
         super(Void.class, environment);
+        this.timeProvider = timeProvider;
     }
 
     @Override
-    protected CoursesByInstitutionResponse processInput(final Void input,
-                                                        final RequestInfo requestInfo,
-                                                        final Context context) {
+    protected Course[] processInput(final Void input,
+                                    final RequestInfo requestInfo,
+                                    final Context context) {
+
+        final Optional<Integer> institutionCode = getInstitutionCodeOfCurrentlyLoggedInUser(requestInfo);
 
-        final String institutionCode = "dummy";
-        logger.debug("Fetching courses by institution '{}'", institutionCode);
+        if (institutionCode.isPresent()) {
+            final String fsConfigString = environment.readEnv(FS_CONFIG_ENV_NAME);
+
+            try {
+                final ObjectMapper objectMapper = getObjectMapper(requestInfo);
+                final FsConfig fsConfig = objectMapper.readValue(fsConfigString.getBytes(StandardCharsets.UTF_8),
+                                                                 FsConfig.class);
+                return fsConfig.getInstitutions().stream()
+                           .filter(inst -> inst.getCode() == institutionCode.get())
+                           .findFirst()
+                           .map(institutionConfig -> {
+                               final CoursesProvider coursesProvider = new CoursesProvider(objectMapper,
+                                                                                           fsConfig.getBaseUri(),
+                                                                                           institutionConfig);
+                               final ZonedDateTime currentTime = timeProvider.getCurrentTime();
+                               return coursesProvider.getCurrentlyTaughtCourses(currentTime.getYear(),
+                                                                                currentTime.getMonthValue());
```

Comment:
I have no idea what is the intention of the code. Refactor using methods with meaningful names. See #3